web/src/engine/websites/RidiBooks.ts

```diff
@@ -0,0 +1,116 @@
+﻿import { Tags } from '../Tags';
+import icon from './RidiBooks.webp';
+import { Chapter, DecoratableMangaScraper, Manga, type MangaPlugin, Page } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchJSON, FetchRequest, FetchWindowScript } from '../FetchProvider';
+
+type APIMangas = {
+    data: {
+        items: APIManga[]
+        pagination: {
+            nextPage: string,
+        }
+    }
+}
+
+type APIManga = {
+    book: {
+        bookId: string,
+        title: string
+        serial: {
+            title: string
+        }
+    }
+}
+
+type APIPages = {
+    data: {
+        pages: {
+            src: string
+        }[]
+    }
+    success: boolean,
+
+}
+
+type ChapterID = {
+    id: string,
+    title: string
+}
+
+type BookDetail = {
+    series_id: string,
+    series_title: string
+};
+
+@Common.ImageAjax()
+export default class extends DecoratableMangaScraper {
+    private apiUrl = 'https://api.ridibooks.com';
+
+    public constructor() {
+        super('ridibooks', 'RidiBooks', 'https://ridibooks.com', Tags.Media.Manhwa, Tags.Language.Korean, Tags.Source.Official);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return new RegExp(`^${this.URI.origin}/books/\\d+`).test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const data = await FetchWindowScript<BookDetail>(new FetchRequest(url), 'bookDetail');
+        return new Manga(this, provider, data.series_id, data.series_title.trim());
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const mangaList : Manga[] = [];
+        let uri = new URL('/v2/category/books', this.apiUrl);
```

Comment:
May explicitly define type, as inferred type is incorrect
`let uri: URL | undefined = ...`

Reply:
I dont get it : `uri ` inferred type is URL to me
![image](https://github.com/manga-download/haruneko/assets/10671518/d98e0675-770b-4aa1-a7db-ed46222e73d4)

Reply:
Yes but later it will assign `undefined` which should not be assigned to a value of type `URL`, the type should be `URL | undefined`